plugins/inputs/host_agent_consumer/host_agent_consumer.go

```diff
@@ -541,30 +559,51 @@ func (h *HostAgent) updateCloudNetworkPorts(netPortMap *CloudMacAddrNetworkMap,
 		json.Unmarshal([]byte(output), &networkPorts)
 
 		log.Printf("I! Loading cloud network names from cloud %s", c.Name)
-		for _, networkPort := range networkPorts.NetworkPorts {
+		for ni := range networkPorts.NetworkPorts {
+			networkPort := &networkPorts.NetworkPorts[ni]
 			if _, ok := (*netPortMap)[networkPort.MacAddress]; !ok {
```

Comment:
The func arg `netPortMap` doesn't need to be a pointer since maps are passed by reference.